tests/layering/node_cmd_test.go

```diff
@@ -0,0 +1,436 @@
+		return nil, fmt.Errorf("could not get config: %w", err)
+	}
+
+	exec, err := remotecommand.NewSPDYExecutor(config, "POST", req.URL())
+	if err != nil {
+		return nil, fmt.Errorf("could not get command executor: %w", err)
+	}
+
+	stdinBuf := bytes.NewBuffer([]byte{})
+	stdoutBuf, stdout := getWriterAndBuffer(runOpts.Stdout)
+	stderrBuf, stderr := getWriterAndBuffer(runOpts.Stderr)
+
+	streamOpts := remotecommand.StreamOptions{
+		Stdout: stdout,
+		Stderr: stderr,
+		Tty:    false,
+	}
+
+	// Wire in stdin using io.TeeReader so its contents are available in the result object.
+	if runOpts.Stdin != nil {
+		streamOpts.Stdin = io.TeeReader(runOpts.Stdin, stdinBuf)
+	}
+
+	// Run the actual command
+	start := time.Now()
+	err = exec.Stream(streamOpts)
+	end := time.Since(start)
+
+	results := CmdResult{
+		Command:  runOpts.Command,
+		Duration: end,
+		Stdin:    stdinBuf.Bytes(),
+		Stdout:   stdoutBuf.Bytes(),
+		Stderr:   stderrBuf.Bytes(),
+	}
+
+	if err != nil {
+		err = &NodeCmdError{
+			CmdResult: results,
+			execErr:   err,
+		}
+	}
+
+	return &results, err
+}
+
+// Creates a new buffer and io.MultiWriter so we can collect stdin / stderr to
+// multiple places simultaneously.
+func getWriterAndBuffer(w io.Writer) (*bytes.Buffer, io.Writer) {
+	buf := bytes.NewBuffer([]byte{})
+	if w == nil {
+		return buf, buf
+	}
+
+	return buf, io.MultiWriter(buf, w)
+}
+
+// Prepends chroot /host onto the command we want to run.
+func getCommandToRun(cmd []string) []string {
+	if strings.HasPrefix(strings.Join(cmd, " "), "chroot /host") {
+		return cmd
+	}
+
+	return append([]string{"chroot", "/host"}, cmd...)
+}
+
+// Determines if the command pod is ready. These checks might be a little
+// heavy-handed, but they work well.
+func isPodReady(pod *corev1.Pod) bool {
+	// Check that the pod is not in the running phase
+	if pod.Status.Phase != corev1.PodRunning {
+		return false
+	}
+
+	// Check all the pod conditions
+	for _, condition := range pod.Status.Conditions {
+		if condition.Type == corev1.PodInitialized && condition.Status != "True" {
+			return false
+		}
+
+		if condition.Type == corev1.PodScheduled && condition.Status != "True" {
+			return false
+		}
+
+		if condition.Type == corev1.PodReady && condition.Status != "True" {
+			return false
+		}
+
+		if condition.Type == corev1.ContainersReady && condition.Status != "True" {
+			return false
+		}
+	}
+
+	// Check all the pod container statuses
+	for _, status := range pod.Status.ContainerStatuses {
+		// The container status is still waiting
+		if status.State.Waiting != nil {
+			return false
+		}
+
+		// The container was terminated
+		if status.State.Terminated != nil {
+			return false
+		}
+
+		// The container isn't running
+		if status.State.Running == nil {
+			return false
+		}
+
+		// This is nil, meaning we haven't started yet
+		if status.Started == nil {
+			return false
+		}
+
+		// We haven't started the container yet
+		if *status.Started != true {
+			return false
```

Comment:
q: probably obvious but why a pointer?

Reply:
The `Started` field is a pointer and one has to deference it in order to check if its true or false. I probably could've done something like:

```golang
if status.Started != nil && *status.Started != true {
  return false
}
```

Reply:
i was thinking the same thing on that one :)